src/iarc7_motion/iarc_tasks/task_utilities/height_holder.py

```diff
@@ -33,34 +35,40 @@ def __init__(self, desired_height = None):
 
         if self._DESIRED_HEIGHT is not None and self._DESIRED_HEIGHT < self._MIN_MANEUVER_HEIGHT:
             raise ValueError('Desired height was below the minimum maneuver height')
-        self._deadzone_activated = False
+        self._height_plan_activated = True
 
-    # uses a p-controller to return a velocity to maintain a height
-    def get_height_hold_response(self, height):
+    # determines whether to set a velocity to maintain height
+    def get_height_hold_response(self, measured_height, predicted_height):
         with self._lock:
+            
+            #Calculates the measured and predicted error
             if self._DESIRED_HEIGHT is None:
                 raise ValueError('No height to hold')
-            self._delta_z = self._DESIRED_HEIGHT - height
+            self._measured_delta_z = self._DESIRED_HEIGHT - measured_height
+            self._predicted_delta_z = self._DESIRED_HEIGHT - predicted_height
 
             if self._DEBUG:
                 msg = Float64ArrayStamped()
                 msg.header.stamp = rospy.Time.now()
-                msg.data = [self._DESIRED_HEIGHT, height, self._delta_z]
+                msg.data = [self._DESIRED_HEIGHT, measured_height, self._measured_delta_z]
                 self._debug_pub.publish(msg)
 
-            # If the drone is in the deadzone area a velocity of zero is commanded
-            # Essentially this is a tolerance region
-            if abs(self._delta_z) < self._DEADZONE:
-                self._deadzone_activated = True
-            elif (self._deadzone_activated == True 
-                     and abs(self._delta_z) > self._DEADZONE
+            reset_z = False
+            # Determines whether height plan should be activated, based on location in relation to the deadzone
+            if (self._height_plan_activated == False):
```

Comment:
Um picky but can these conditionals be cleaned up. maybe like `if not self._height_plan_activated`